packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -1486,7 +1486,13 @@ export class StripeHelper {
   }
 
   /**
-   * Extract invoice details for billing emails
+   * Extract invoice details for billing emails.
+   *
+   * Note that this function throws an error in the following cases:
+   *   - Stripe customer is deleted.
+   *   - No plan in the invoice.
+   *   - No product attached to the plan.
```

Comment:
??? Does Stripe even allow this?

Reply:
I didn't think they do, but its in the code as a type guard check which means the Stripe API types believe it could occur.